### ISSUE TITLE
Fix flaky CI test on Windows

### ISSUE THREAD

**Issue by martincostello** [domaindrivendev/Swashbuckle.AspNetCore#2836]
The tests on Windows are being a bit flaky due to some file copy conflicts during the build of the test projects.

Need to do _something_ to resolve that.

**Comment by martincostello** [comment]
[Example](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/actions/runs/8829747109/job/24241507993?pr=2677#step:5:505):

```console
  Failed Swashbuckle.AspNetCore.IntegrationTests.SwaggerIntegrationTests.SwaggerEndpoint_ReturnsValidSwaggerJson(startupType: typeof(OAuth2Integration.Startup), swaggerRequestUri: "/resource-server/swagger/v1/swagger.json") [362 ms]
  Error Message:
   System.IO.IOException : The process cannot access the file 'D:\a\Swashbuckle.AspNetCore\Swashbuckle.AspNetCore\artifacts\bin\Swashbuckle.AspNetCore.IntegrationTests\release\tempkey.rsa' because it is being used by another process.
  Stack Trace:
     at Microsoft.Win32.SafeHandles.SafeFileHandle.CreateFile(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategyCore(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.WriteAllText(String path, String contents)
   at Microsoft.Extensions.DependencyInjection.IdentityServerBuilderExtensionsCrypto.AddDeveloperSigningCredential(IIdentityServerBuilder builder, Boolean persistKey, String filename, RsaSigningAlgorithm signingAlgorithm)
   at OAuth2Integration.Startup.ConfigureServices(IServiceCollection services) in /_/test/WebSites/OAuth2Integration/Startup.cs:line 26
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.InvokeCore(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass9_0.<Invoke>g__Startup|0(IServiceCollection serviceCollection)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass15_0.<BuildStartupServicesFilterPipeline>g__RunPipeline|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.Invoke(Object instance, IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConfigureServicesBuilder.<>c__DisplayClass8_0.<Build>b__0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.StartupLoader.ConfigureServicesDelegateBuilder`1.<>c__DisplayClass14_0.<ConfigureServices>g__ConfigureServicesWithContainerConfiguration|0(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at Microsoft.AspNetCore.TestHost.TestServer..ctor(IWebHostBuilder builder, IFeatureCollection featureCollection)
   at Microsoft.AspNetCore.TestHost.TestServer..ctor(IWebHostBuilder builder)
   at Swashbuckle.AspNetCore.IntegrationTests.TestSite.BuildServer() in /_/test/Swashbuckle.AspNetCore.IntegrationTests/TestSite.cs:line 29
   at Swashbuckle.AspNetCore.IntegrationTests.TestSite.BuildClient() in /_/test/Swashbuckle.AspNetCore.IntegrationTests/TestSite.cs:line 34
   at Swashbuckle.AspNetCore.IntegrationTests.SwaggerIntegrationTests.SwaggerEndpoint_ReturnsValidSwaggerJson(Type startupType, String swaggerRequestUri) in /_/test/Swashbuckle.AspNetCore.IntegrationTests/SwaggerIntegrationTests.cs:line 34
--- End of stack trace from previous location ---
```

**Comment by martincostello** [comment]
Updating this target to do a few retries if there's an file write conflict should fix _most_ cases of that problem:

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/5b62825303eeaeae96d96a4ca6bd2ca4b968c435/Directory.Build.targets#L25

**Comment by martincostello** [comment]
I think this is resolved now since #2848.
